### PR TITLE
feat: add new `ssg.strict` config

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

Add a new `ssg.strict` config.

```js
export default {
  ssg: {
    strict: true,
  },
};
```

## Related Issue

See https://github.com/web-infra-dev/rspress/issues/1317 for more background.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Review by Timeless0911 — APPROVED** [review]

File: packages/core/src/node/build.ts [hunk 1]
```diff
@@ -80,18 +82,25 @@ export async function renderPages(
     // 2. ssr bundle render failed
     // 3. ssg is disabled
     let render = null;
-    if (enableSSG) {
+    if (ssgConfig) {
       try {
         const { default: ssrExports } = await import(
           pathToFileURL(ssrBundlePath).toString()
         );
         ({ render } = ssrExports as SSRBundleExports);
```
Comment (zoolsher) [inline packages/core/src/node/build.ts:90]: This render function gets invoked later to render the page dom. There is also a try catch. Should we also add a throw error there?

Comment (chenjiahan) [reply packages/core/src/node/build.ts:90]: Nice catch, I missed that.

Updated 😄

File: packages/core/src/node/build.ts [hunk 2]
```diff
@@ -129,10 +138,14 @@ export async function renderPages(
             try {
               ({ appHtml } = await render(routePath, helmetContext.context));
             } catch (e) {
+              if (typeof ssgConfig === 'object' && ssgConfig.strict) {
```
Comment (zoolsher) [inline packages/core/src/node/build.ts:141]: Do you think we should provide additional context to assist users with debugging? like add `routePath` or more to the error.message

Comment (chenjiahan) [reply packages/core/src/node/build.ts:141]: Sure 👍  the error logs has been updated:

Screenshot 2024-08-14 at 11 48 44

Comment (zoolsher) [reply packages/core/src/node/build.ts:144]: Cool. Really helpful error diagnostic

**Review by zoolsher — APPROVED** [review]
LGTM

**Pull request merged**